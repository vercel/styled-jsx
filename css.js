function notTranspiledError(name) {
  throw new Error(
    'styled-jsx/css: if you are getting this error it means that your `' +
      name +
      '` tagged template literals were not transpiled.'
  )
}
function css() {
  notTranspiledError('css')
}
css.global = function() {
  notTranspiledError('global')
}
css.resolve = function() {
  notTranspiledError('resolve')
}
module.exports = css
module.exports.global = css.global
module.exports.resolve = css.resolve
