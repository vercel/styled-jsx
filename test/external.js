// Packages
import test from 'ava'

// Ours
import plugin from '../src/babel'
import _transform, { transformSource as _transformSource } from './_transform'

const transform = (file, opts = {}) =>
  _transform(file, {
    plugins: [[plugin, opts]]
  })

const transformSource = (src, opts = {}) =>
  _transformSource(src.trim(), {
    plugins: [[plugin, opts]],
    ...opts
  })

test('transpiles external stylesheets', async t => {
  const { code } = await transform('./fixtures/styles.js')
  t.snapshot(code)
})

test('(optimized) transpiles external stylesheets', async t => {
  const { code } = await transform('./fixtures/styles.js', {
    optimizeForSpeed: true
  })
  t.snapshot(code)
})

test('transpiles external stylesheets (CommonJS modules)', async t => {
  const { code } = await transform('./fixtures/styles2.js')
  t.snapshot(code)
})

test('(optimized) transpiles external stylesheets (CommonJS modules)', async t => {
  const { code } = await transform('./fixtures/styles2.js', {
    optimizeForSpeed: true
  })
  t.snapshot(code)
})

test('does not transpile non-styled-jsx tagged teplate literals', async t => {
  const { code } = await transform(
    './fixtures/not-styled-jsx-tagged-templates.js'
  )
  t.snapshot(code)
})

test('throws when using `this.something` in external stylesheets', async t => {
  const { message } = await t.throws(
    transform('./fixtures/styles-external-invalid.js')
  )
  t.regex(message, /this\.props/)
})

test('throws when referring an undefined value in external stylesheets', async t => {
  const { message } = await t.throws(
    transform('./fixtures/styles-external-invalid2.js')
  )
  t.regex(message, /props\.color/)
})

test('use external stylesheets', async t => {
  const { code } = await transform('./fixtures/external-stylesheet.js')
  t.snapshot(code)
})

test('use external stylesheets (multi-line)', async t => {
  const { code } = await transform(
    './fixtures/external-stylesheet-multi-line.js'
  )
  t.snapshot(code)
})

test('use external stylesheets (global only)', async t => {
  const { code } = await transform('./fixtures/external-stylesheet-global.js')
  t.snapshot(code)
})

test('injects JSXStyle for nested scope', async t => {
  const { code } = await transformSource(`
    import css from 'styled-jsx/css'

    function test() {
      css.resolve\`div { color: red }\`
    }
  `)
  t.snapshot(code)
})
