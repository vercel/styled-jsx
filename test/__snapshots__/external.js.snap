// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(optimized) transpiles external stylesheets (CommonJS modules) 1`] = `
"const _defaultExport = ['div.jsx-2292456818{font-size:3em;}'];
_defaultExport.__hash = '2292456818';


module.exports = _defaultExport;"
`;

exports[`(optimized) transpiles external stylesheets 1`] = `
"import _JSXStyle from 'styled-jsx/style';

import colors, { size } from './constants';
const color = 'red';

const bar = ['div.jsx-2141779268{font-size:3em;}'];

bar.__hash = '2141779268';
const baz = ['div{font-size:3em;}'];

baz.__hash = '2141779268';
a.__hash = '262929833';
const a = [\`div{font-size:\${size}em;}\`];

export const uh = bar;

export const foo = [\`div.jsx-2299908427{color:\${color};}\`];

foo.__hash = '2299908427';
({
  styles: <_JSXStyle styleId={\\"1329679275\\"} css={[\`div.jsx-1329679275{color:\${colors.green.light};}\`, 'a.jsx-1329679275{color:red;}']} />,
  className: 'jsx-1329679275'
});

const b = {
  styles: <_JSXStyle styleId={\\"1329679275\\"} css={[\`div.jsx-1329679275{color:\${colors.green.light};}\`, 'a.jsx-1329679275{color:red;}']} />,
  className: 'jsx-1329679275'
};

const dynamic = colors => {
  const b = {
    styles: <_JSXStyle styleId={\\"3325296745\\"} css={[\`div.__jsx-style-dynamic-selector{color:\${colors.green.light};}\`, 'a.__jsx-style-dynamic-selector{color:red;}']} dynamic={[colors.green.light]} />,
    className: _JSXStyle.dynamic([['3325296745', [colors.green.light]]])
  };
};

export default {
  styles: <_JSXStyle styleId={\\"3290112549\\"} css={['div.jsx-3290112549{font-size:3em;}', \`p.jsx-3290112549{color:\${color};}\`]} />,
  className: 'jsx-3290112549'
};"
`;

exports[`does not transpile non-styled-jsx tagged teplate literals 1`] = `
"import css from 'hell';

const color = 'red';

const bar = css\`
  div {
    font-size: 3em;
  }
\`;
export const uh = bar;

export const foo = css\`div { color: \${color}}\`;

export default css\`
  div {
    font-size: 3em;
  }
  p {
    color: \${color};
  }
\`;

const Title = styled.h1\`
  color: red;
  font-size: 50px;
\`;

const AnotherTitle = Title.extend\`color: blue;\`;

export const Component = () => <AnotherTitle>My page</AnotherTitle>;"
`;

exports[`injects JSXStyle for nested scope 1`] = `
"import _JSXStyle from 'styled-jsx/style';


function test() {
  ({
    styles: <_JSXStyle styleId={\\"2886504620\\"} css={\\"div.jsx-2886504620{color:red;}\\"} />,
    className: 'jsx-2886504620'
  });
}"
`;

exports[`transpiles external stylesheets (CommonJS modules) 1`] = `
"const _defaultExport = new String('div.jsx-2292456818{font-size:3em;}');

_defaultExport.__hash = '2292456818';


module.exports = _defaultExport;"
`;

exports[`transpiles external stylesheets 1`] = `
"import _JSXStyle from 'styled-jsx/style';

import colors, { size } from './constants';
const color = 'red';

const bar = new String('div.jsx-2141779268{font-size:3em;}');

bar.__hash = '2141779268';
const baz = new String('div{font-size:3em;}');

baz.__hash = '2141779268';
a.__hash = '262929833';
const a = new String(\`div{font-size:\${size}em;}\`);

export const uh = bar;

export const foo = new String(\`div.jsx-2299908427{color:\${color};}\`);

foo.__hash = '2299908427';
({
  styles: <_JSXStyle styleId={\\"1329679275\\"} css={\`div.jsx-1329679275{color:\${colors.green.light};}a.jsx-1329679275{color:red;}\`} />,
  className: 'jsx-1329679275'
});

const b = {
  styles: <_JSXStyle styleId={\\"1329679275\\"} css={\`div.jsx-1329679275{color:\${colors.green.light};}a.jsx-1329679275{color:red;}\`} />,
  className: 'jsx-1329679275'
};

const dynamic = colors => {
  const b = {
    styles: <_JSXStyle styleId={\\"3325296745\\"} css={\`div.__jsx-style-dynamic-selector{color:\${colors.green.light};}a.__jsx-style-dynamic-selector{color:red;}\`} dynamic={[colors.green.light]} />,
    className: _JSXStyle.dynamic([['3325296745', [colors.green.light]]])
  };
};

export default {
  styles: <_JSXStyle styleId={\\"3290112549\\"} css={\`div.jsx-3290112549{font-size:3em;}p.jsx-3290112549{color:\${color};}\`} />,
  className: 'jsx-3290112549'
};"
`;

exports[`use external stylesheets (global only) 1`] = `
"import _JSXStyle from 'styled-jsx/style';
import styles, { foo as styles3 } from './styles';

const styles2 = require('./styles2');

export default (() => <div>
    <p>test</p>
    <div>woot</div>
    <p>woot</p>
    <_JSXStyle styleId={styles2.__hash} css={styles2} />
    <_JSXStyle styleId={styles3.__hash} css={styles3} />
    <_JSXStyle styleId={styles.__hash} css={styles} />
  </div>);"
`;

exports[`use external stylesheets (multi-line) 1`] = `
"import _JSXStyle from 'styled-jsx/style';
import styles from './styles';

export default (() => <div className={\`jsx-\${styles.__scopedHash}\`}>
    <p className={\`jsx-\${styles.__scopedHash}\`}>test</p>
    <_JSXStyle styleId={styles.__scopedHash} css={styles.__scoped} />
  </div>);"
`;

exports[`use external stylesheets 1`] = `
"import _JSXStyle from 'styled-jsx/style';
import styles from './styles';
const styles2 = require('./styles2');
import { foo as styles3 } from './styles';

export default (() => <div className={'jsx-1646697228 ' + \`jsx-\${styles3.__scopedHash} jsx-\${styles.__scopedHash}\`}>
    <p className={'jsx-1646697228 ' + \`jsx-\${styles3.__scopedHash} jsx-\${styles.__scopedHash}\` + ' ' + 'foo'}>test</p>
    <p className={'jsx-1646697228 ' + \`jsx-\${styles3.__scopedHash} jsx-\${styles.__scopedHash}\`}>woot</p>
    <_JSXStyle styleId={styles2.__hash} css={styles2} />
    <_JSXStyle styleId={styles3.__scopedHash} css={styles3.__scoped} />
    <div className={'jsx-1646697228 ' + \`jsx-\${styles3.__scopedHash} jsx-\${styles.__scopedHash}\`}>woot</div>
    <_JSXStyle styleId={\\"1646697228\\"} css={\\"p.jsx-1646697228{color:red;}div.jsx-1646697228{color:green;}\\"} />
    <_JSXStyle styleId={styles.__scopedHash} css={styles.__scoped} />
  </div>);

export const Test = () => <div className={'jsx-1646697228 ' + \`jsx-\${styles3.__scopedHash}\`}>
    <p className={'jsx-1646697228 ' + \`jsx-\${styles3.__scopedHash}\` + ' ' + 'foo'}>test</p>
    <p className={'jsx-1646697228 ' + \`jsx-\${styles3.__scopedHash}\`}>woot</p>
    <_JSXStyle styleId={styles3.__scopedHash} css={styles3.__scoped} />
    <div className={'jsx-1646697228 ' + \`jsx-\${styles3.__scopedHash}\`}>woot</div>
    <_JSXStyle styleId={\\"1646697228\\"} css={\\"p.jsx-1646697228{color:red;}div.jsx-1646697228{color:green;}\\"} />
  </div>;"
`;
